aten/src/ATen/detail/HIPHooksInterface.cpp

```diff
@@ -0,0 +1,35 @@
+#include <ATen/detail/HIPHooksInterface.h>
+
+#include <c10/util/Exception.h>
+
+#include <cstddef>
+#include <memory>
+#include <mutex>
+
+namespace at {
+namespace detail {
+
+const HIPHooksInterface& getHIPHooks() {
+  static std::unique_ptr<HIPHooksInterface> hip_hooks;
+  // NB: The once_flag here implies that if you try to call any HIP
+  // functionality before libATen_cuda.so is loaded, HIP is permanently
+  // disabled for that copy of ATen.  In principle, we can relax this
+  // restriction, but you might have to fix some code.  See getVariableHooks()
```

Comment:
you don't need to change this, but I bet you this comment lasts longer than getVariableHooks exists :).